modules/core/keeper/msg_server_test.go

```diff
@@ -827,7 +831,74 @@ func (suite *KeeperTestSuite) TestUpdateClientParams() {
 }
 
 // TestIBCSoftwareUpgrade tests the IBCSoftwareUpgrade rpc handler
-func (*KeeperTestSuite) TestIBCSoftwareUpgrade() {
+func (suite *KeeperTestSuite) TestIBCSoftwareUpgrade() {
+	var (
+		expError error
+		msg      *clienttypes.MsgIBCSoftwareUpgrade
+	)
+	testCases := []struct {
+		name     string
+		malleate func()
+		expPass  bool
```

Comment:
same as other PR, we can refactor this to be `expError` and use `expPass := expError == nil` in the test body! 👍